resources/ext.wikibase.facetedsearch.js

```diff
@@ -0,0 +1,78 @@
+let input;
+
+/**
+ * Main entry point for the JavaScript code.
+ */
+function init() {
+	const facets = document.querySelector( '.wikibase-faceted-search__facets' );
+	if ( !facets ) {
+		return;
+	}
+
+	input = document.querySelector( '#searchText > input' );
+
+	facets.addEventListener( 'change', onFacetsChange );
+}
+
+/**
+ * Handles the change event for any input elements in the facets.
+ *
+ * @param {Event} event
+ */
+function onFacetsChange( event ) {
+	const facet = event.target.closest( '.wikibase-faceted-search__facet' );
+	// TODO: Support range facets
+	if ( event.target.classList.contains( 'cdx-checkbox__input' ) ) {
+		submitSearchForm( buildQueryString( input.value, facet ) );
+	}
+}
+
+/**
+ * Builds a new query string from the given old query and facet.
+ *
+ * @param {string} oldQuery The original query string.
+ * @param {HTMLElement} facet The facet element.
+ *
+ * @return {string} The new query string.
+ */
+function buildQueryString( oldQuery, facet ) {
+	if ( !facet ) {
+		return oldQuery;
+	}
+
+	const propertyId = facet.dataset.propertyId;
+	if ( !propertyId ) {
+		return oldQuery;
+	}
+
+	// Remove existing facet filters for the same property ID
+	const queries = oldQuery.split( /\s+/ ).filter(
+		( item ) => !new RegExp( `^(haswbfacet|\\-haswbfacet):${ propertyId }(=|>=|<=)` ).test( item )
+	);
+
+	// Add selected facet items to the query string
+	[ ...facet.querySelectorAll( '.wikibase-faceted-search__facet-item' ) ].forEach( ( facetItem ) => {
+		const checkbox = facetItem.querySelector( '.cdx-checkbox__input' );
+		if ( !checkbox || !checkbox.checked || !checkbox.value ) {
+			return;
+		}
+		// TODO: Support other operators
+		// TODO: Support OR values
+		queries.push( `haswbfacet:${ propertyId }=${ checkbox.value }` );
+	} );
+
+	return queries.join( ' ' );
+}
```

Comment:
Some functions can be extracted

If you find yourself writing comments for blocks of code in a function, that is a signal that those blocks maybe should be their own functions. Comments are rarely needed if you do one thing on one level of attraction per function and you name your functions well.

Reply:
Btw, function names behave oposite of variables wrt optimal length. Functions in small scopes often have long names.